sf-stream/src/protos/extractor.proto

```diff
@@ -0,0 +1,290 @@
+
+message DeleteModule {
+  string address = 1;
+  bytes state_key_hash = 2;
+  MoveModuleId module = 3;
+}
+
+message DeleteResource {
+  string address = 1;
+  bytes state_key_hash = 2;
+  MoveStructTag resource = 3;
+}
+
+message DeleteTableItem {
+  bytes state_key_hash = 1;
+  string handle = 2;
+  string key = 3;
+  DeleteTableData data = 4;
+}
+
+message DeleteTableData {
+  string key = 1;
+  string key_type = 2;
+}
+
+message WriteModule {
+  string address = 1;
+  bytes state_key_hash = 2;
+  string data = 3;
+}
+
+message WriteResource {
+  string address = 1;
+  bytes state_key_hash = 2;
+  MoveResource data = 3;
+}
+
+message WriteTableData {
+  string key = 1;
+  string key_type = 2;
+  string value = 3;
+  string value_type = 4;
+}
+
+message WriteTableItem {
+  bytes state_key_hash = 1;
+  string handle = 2;
+  string key = 3;
+  WriteTableData data = 4;
+}
+
+message TransactionPayload {
+
+  enum PayloadType {
+    SCRIPT_FUNCTION_PAYLOAD = 0;
+    SCRIPT_PAYLOAD = 1;
+    MODULE_BUNDLE_PAYLOAD = 2;
+    WRITE_SET_PAYLOAD = 3;
+  }
+
+  PayloadType type = 1;
+  oneof payload {
+    ScriptFunctionPayload script_function_payload = 2;
+    ScriptPayload script_payload = 3;
+    ModuleBundlePayload module_bundle_payload = 4;
+    WriteSetPayload write_set_payload = 5;
+  }
+}
+
+message ScriptFunctionPayload {
+  ScriptFunctionId function = 1;
+  repeated string type_arguments = 2;
+  repeated string arguments = 3;
+}
+
+message ScriptPayload {
+  string code = 1;
+  repeated string type_arguments = 2;
+  repeated string arguments = 3;
+}
+
+message ModuleBundlePayload {
+  repeated string modules = 1;
+}
+
+message WriteSetPayload {
+  WriteSet write_set = 1;
+}
+
+message ScriptFunctionId {
+  MoveModuleId module = 1;
+  string name = 2;
+}
+
+message MoveResource {
+  MoveStructTag type = 1;
+  string data = 2;
+}
+
+message MoveModuleId {
+  string address = 1;
+  string name = 2;
+}
+
+message MoveStructTag {
+  string address = 1;
+  string module = 2;
+  string name = 3;
+  repeated string generic_type_params = 4;
+}
+
+message Signature {
+
+  enum SignatureType {
+    ED = 0;
+    MULTI_ED = 1;
+    MULTI_AGENT = 2;
+  }
+
+  SignatureType type = 1;
+  oneof signature {
+    Ed25519Signature ed = 2;
+    MultiEd25519Signature multi_ed = 3;
+    MultiAgentSignature multi_agent = 4;
+  }
+}
+
+message Ed25519Signature {
+  string public_key = 1;
+  string signature = 2;
+}
+
+message MultiEd25519Signature {
+  repeated string public_keys = 1;
+  repeated string signatures = 2;
+  uint32 threshold = 3;
+  string bitmap = 4;
+}
+
+message MultiAgentSignature {
+  AccountSignature sender = 1;
+  repeated string secondary_signer_addresses = 2;
+  repeated AccountSignature secondary_signers = 3;
+}
+
+message AccountSignature {
+
+  enum AccountSignatureType {
+    ED = 0;
+    MULTI_ED = 1;
+  }
+
+  AccountSignatureType type = 1;
+  oneof signature {
+    Ed25519Signature ed = 2;
+    MultiEd25519Signature multi_ed = 3;
+  }
+}
```

Comment:
This is strange, which one is multisigner vs multi agent?

Reply:
updated :-)